### PR TITLE
Result plugins execution order support [v0]

### PR DESCRIPTION
**Pull request by clebergnu**
This is an early (v0, missing docs and more tests) ordered execution support for result plugins.

Also, it turns the archive (`--archive`, `-z`) support into a plugin.

While this achieves the actual use cases discussed in the mailling list RFCs and the following trello cards:

https://trello.com/c/oWzrV48E/837-execution-order-support-in-plugins
https://trello.com/c/MjPJel8u/838-job-result-archive-support-turn-into-a-plugin

Support for more generic execution order support (all plugin types) is highly desirable, as is more tests and documentation.

For reviewers: please check if the use cases are indeed covered.  A slight different implementation should come up in v1.

### CODE REVIEW

File: etc/avocado/avocado.conf [hunk 1]
```diff
@@ -85,3 +85,7 @@ skip_broken_plugin_notification = []
 # options or test types).
 # The keyword "@DEFAULT" will be replaced with all available unused loaders.
 loaders = ['file', '@DEFAULT']
+
+[plugins.result]
+# Support for ordering result plugins *only* at this version
```
Comment (adereis) [inline etc/avocado/avocado.conf]: I guess you mean "only result plugins" and not "only at this version". Please rephrase to remove the ambiguity.

Comment (clebergnu) [inline etc/avocado/avocado.conf]: Yep, that was the idea.  As mentioned, I'll probably lift the limitation on v1, so there may be no need to rephrase it.

Comment (clebergnu) [inline etc/avocado/avocado.conf]: BTW, I forgot to add the `archive.py` plugin to the last commit.  Force pushing that last one.

File: avocado/core/dispatcher.py [hunk 2]
```diff
@@ -51,10 +51,18 @@ def plugin_type(self):
         else:
             return self.namespace
 
+    def fully_qualified_name(self, extension):
```
Comment (ldoktor) [inline avocado/core/dispatcher.py]: Actually thinking about it a bit more, how about supporting those directly from each extension? The extension would be responsible to provide `type` (without the namespace prefix, that's the only little issue) and `fqn`.

Comment (ldoktor) [inline avocado/core/dispatcher.py]: Also if we want to keep this in dispatcher itself, I think we should use a better names as it works for the extensions. Something like `get_extension_type` and `get_extension_fqn`. But I still think it'd be better to move this into the extensions viz previous comment.

Comment (ldoktor) [inline avocado/core/dispatcher.py]: OK, reading it a bit more, the `plugin_type` is actually dispatcher's method and the `fqn` is also influenced by that. I need to think about it a bit more, but currently the names are a bit misleading as they combine dispatcher and extension properties.

File: avocado/core/dispatcher.py [hunk 3]
```diff
@@ -133,8 +133,25 @@ class ResultDispatcher(Dispatcher):
...
+            for ext in self.extensions:
+                if name == ext.name:
+                    ordered.append(ext)
         for ext in self.extensions:
+            if ext not in ordered:
+                ordered.append(ext)
+        return ordered
+
+    def map_method(self, method_name, result, job):
+        for ext in self.ordered_extensions():
```
Comment (ldoktor) [inline avocado/core/dispatcher.py]: well iterator would be probably better, but sure, it serves the purpose...

**Comment by clebergnu** [comment]
Closing in favor of #1565

**Pull request closed without merging**